src/sentry/search/events/builder.py

```diff
@@ -1945,7 +1952,7 @@ def _environment_filter_converter(self, search_filter: SearchFilter) -> Optional
         values_set = set(value if isinstance(value, (list, tuple)) else [value])
         # sorted for consistency
         values = sorted(
-            self.config.resolve_value(f"{value}") if value else 0 for value in values_set
+            self._resolve_environment_filter_value(value) if value else 0 for value in values_set
```

Comment:
Not sure if we need `if value else 0` since I don't know what case would cause that condition

Reply:
Spoke to Will about this yesterday, it's to handle empty strings in the search filter. In the array case, what we have here is fine. It needs to be handled better for non-"IN" filter type cases separately (not in the scope of this PR)